lib/perl/Genome/Sys/Lock.pm

```diff
@@ -151,30 +159,63 @@ sub with_default_lock_resource_args {
 
     $args{block_sleep} = 60 unless defined $args{block_sleep};
     $args{max_try} = 7200 unless defined $args{max_try};
+    $args{scope} = $ENV{GENOME_SYS_LOCK_DEFAULT_SCOPE} unless defined $args{scope};
     $args{wait_announce_interval} = 0 unless defined $args{wait_announce_interval};
 
     return %args;
 }
 
-my @backends = ('Genome::Sys::FileLock');
-
 sub is_mandatory {
     my $backend = shift;
     return $backend->is_mandatory();
 }
 
+my $backends = {};
 sub backends {
-    return @backends;
+    my $scope = shift;
+
+    if (!defined($scope)) {
+        return %$backends;
+
+    } elsif ($scope eq 'any') {
+        return map {backends($_)} scopes();
+
+    } elsif (exists $backends->{$scope}) {
+        return @{$backends->{$scope}};
+
+    } else {
+        Carp::confess(sprintf(
+                'Unknown locking scope (%s) -- expected something in [%s].',
+                $scope, join(' ', scopes())));
+    }
 }
 
-sub add_backend {
-    my ($class, $backend) = @_;
-    push @backends, $backend;
+sub scopes {
+    return keys %$backends;
 }
 
-sub remove_backend {
-    my ($class, $backend) = @_;
-    @backends = grep { $_ ne $backend } @backends;
+sub clear_backends {
+    $backends = {};
+}
+
+sub set_backends {
+    my $class = shift;
+    my %new_backends = @_;
+
+    $class->clear_backends();
+    for my $scope (keys %new_backends) {
+        for my $backend (@{$new_backends{$scope}}) {
+            $class->add_backend($scope, $backend);
+        }
+    }
+}
+
+sub add_backend {
+    my ($class, $scope, $backend) = @_;
+    if ($scope eq 'any') {
```

Comment:
Oops.  For a while, I was treating the scope `any` as a special scope, so I needed to check for its validity.  I think we should just remove the check and leave it to configuration to decide what the scopes are.